### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by AnicetFantomas**
In this pull request we performed :

- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -9,8 +9,8 @@ class Post < ApplicationRecord
 
   after_save :update_posts_counter
 
-  def recent_comments
-    comments.order(created_at: :desc).limit(5)
+  def most_recent_five_comments
+    comments.includes(:user).order(created_at: :desc).limit(5)
```
Comment (ShadyShawkat) [inline app/models/post.rb:13]: - [ ] Kindly make sure to solve the N+1 problem from the posts controller and the user controller not in the models, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.) requirement.

File: spec/integration/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show', type: :feature do
+  describe 'Post' do
+    before(:each) do
+      @first_user = User.create(name: 'Tom', photo: 'margaret.jpeg', bio: 'Teacher from Mexico.', posts_counter: 0)
+      @first_user.save!
```
Comment (ShadyShawkat) [inline spec/integration/post_show_spec.rb:7]: - [ ] Please make sure that the tests run successfully and pass. Currently, there're tests that fail, check the attached below.
image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @AnicetFantomas,** 

### **Great work on making the changes 👏** 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## **Suggested changes**
_Check the comments under the review._

- **Nice job my friend, but there are still some tests that are failing so please fix them I attached a screenshot to help you.**

Capture d’écran 2022-07-27 110618
Capture d’écran 2022-07-27 110651



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AnicetFantomas, @YannickZahinda and @white3d,

✔️ All integration tests are passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

## NB: You should consider using the default `.gitgnore` file that comes with the `rails new` command to avoid committing unnecessary files to your repository. It is set up to ignore certain files in the `tmp` directory while not ignoring the directory completely.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @AnicetFantomas 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**